File: src/main/java/net/reflact/engine/quest/QuestData.java
package net.reflact.engine.quest;

import java.util.List;

public class QuestData {
    public String id;
    public String name;
    public List<String> description;
    public List<Stage> stages;
    
    public static class Stage {
        public String description;
        public String type; // TALK, KILL, FETCH
        public String target; // NpcID, EntityType, ItemID
        public int count;
    }
}

File: src/main/java/net/reflact/engine/quest/QuestData.kt
package net.reflact.engine.quest

class QuestData {
    var id: String = ""
    var name: String = ""
    var description: List<String> = ArrayList()
    var stages: List<Stage> = ArrayList()

    class Stage {
        var description: String = ""
        var type: String = "" // TALK, KILL, FETCH
        var target: String = "" // NpcID, EntityType, ItemID
        var count: Int = 0
    }
}
